apache/couchdb-fauxton
# [partitioned-dbs] Updates to doc editor

## Overview

Third in a series of PRs that will be submitted in support of the new user-defined partitioned databases feature (https://github.com/apache/couchdb/pull/1605).

Summary:
- Creating a new doc: auto-generated doc ID includes partition key
- Cloning a doc: auto-generated ID has the same partition key as the original doc.
- Clicking the cancel button will navigate back to All Docs and preserve the previously selected partition key.
  - Same is true even after cloning a doc

## Testing recommendations

Using a partitioned database:

- Auto-generated doc ID
  - Go to All Documents, and don't select a partition key.
  - Click on Create Document
    - Verify new ID starts with ':'
  - Go back to All Documents and input a key in the Partition Key Selector. Press Enter to set it.
  - Click on Create Document
    - Verify new ID starts with the partition key you selected
  - Click Cancel
    - Verify you are redirected to All Documents and the same partition key is still selected

- Cloning
  - Go to All Documents and click on an existing doc
  - In the code editor, click on Clone Document
    - Verify suggested new ID starts with the same key as the original doc
  - Click on Clone Document
    - Verify the editor loads the cloned document
  - Click on Cancel
    - Verify you are redirected to All Documents

- Cancel button redirects to previous page
  - Click on a view
  - Select a partition key at the top. Press Enter to set it.
   - Note: results are not being filtered yet but you should see the URL is updated
  - Click on any doc to open the editor
  - Click Cancel
     - Verify you are redirected to the same view, and the same partition key is still selected

## Related Pull Requests

https://github.com/apache/couchdb/pull/1605

## Checklist
